src/sxapi/base.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/python
+# coding: utf8
+
+import requests
+
+PUBLIC_API_V2 = "https://api.smaxtec.com/api/v2"
+INTEGRATION_API_V2 = "https://api.smaxtec.com/integration/v2"
+
+
+class BaseAPI(object):
+    def __init__(self, base_url, email=None, password=None, api_key=None):
+        """Initialize a new base low level API client instance.
+        """
+        self.api_base_url = base_url.rstrip("/")
+        self.email = email
+        self.password = password
+        self.api_key = api_key
+        self._session_key = None
+        self._session = None
+
+    @property
+    def session(self):
+        """
+            Geneate a new HTTP session on the fly and login.
+        """
+        if not self._session:
+            self._session = requests.Session()
+        # check login
+        if not self._login():
```

Comment:
further:
self._login() never returns False, so the ValueError cannot be raised.
we might do just 
```self._login() ```
and ignore the return value.

Reply:
I think just self._login() should be fine.